algorithms/symmetry/__init__.py

```diff
@@ -156,16 +156,27 @@ def _normalise(self, method):
         elif method == "ml_aniso":
             normalise = self.ml_aniso_normalisation
 
-        for i in range(int(flex.max(self.dataset_ids) + 1)):
-            logger.info("\n" + "-" * 80 + "\n")
-            logger.info("Normalising intensities for dataset %i\n" % (i + 1))
-            intensities = self.intensities.select(self.dataset_ids == i)
-            if i == 0:
-                normalised_intensities = normalise(intensities)
-            else:
-                normalised_intensities = normalised_intensities.concatenate(
-                    normalise(intensities)
-                )
+        try:
+            for i in range(int(flex.max(self.dataset_ids) + 1)):
+                logger.info("\n" + "-" * 80 + "\n")
+                logger.info("Normalising intensities for dataset %i\n" % (i + 1))
+                intensities = self.intensities.select(self.dataset_ids == i)
+                if i == 0:
+                    normalised_intensities = normalise(intensities)
+                else:
+                    normalised_intensities = normalised_intensities.concatenate(
+                        normalise(intensities)
+                    )
+        except RuntimeError:
```

Comment:
same as above - I assume we expect the `RuntimeError` to happen in `normalise()` rather than `.select` or `.concatenate` (or `flex.max`)

Reply:
Quite right.  Thank you.